src/test/java/net/openhft/chronicle/queue/bench/MethodReaderSkipBenchmark.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2014-2020 chronicle.software
+ *
+ *       https://chronicle.software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.openhft.chronicle.queue.bench;
+
+import net.openhft.chronicle.bytes.MethodReader;
+import net.openhft.chronicle.core.io.IORuntimeException;
+import net.openhft.chronicle.core.io.IOTools;
+import net.openhft.chronicle.core.util.NanoSampler;
+import net.openhft.chronicle.jlbh.JLBH;
+import net.openhft.chronicle.jlbh.JLBHOptions;
+import net.openhft.chronicle.jlbh.JLBHTask;
+import net.openhft.chronicle.jlbh.TeamCityHelper;
+import net.openhft.chronicle.queue.ChronicleQueue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+
+public class MethodReaderSkipBenchmark implements JLBHTask {
+    private static final String PAYLOAD = "this is a string which is here to provide a payload for the method reader to skip over";
+    private static int iterations;
+    private JLBH jlbh;
+    private ChronicleQueue queue;
+    private NanoSampler noopSampler;
+    private NanoSampler wroteSampler;
+    private AnInterface writer;
+    private MethodReader reader;
+    private int counter = 0;
+
+    public static void main(String[] args) {
+        iterations = Integer.getInteger("benchmarkIterations", 100_000);
+        System.out.println("Iterations: " + iterations);
+        int throughput = Integer.getInteger("benchmarkThroughput", 20_000);
+        System.out.println("Throughput: " + throughput);
+
+        // disable as otherwise single GC event skews results heavily
```

Comment:
This comment seems orphaned,  I think I recall it being about coordinated omission

Reply:
good point. Removed. The joy of copy/paste!